### PR TITLE
chore: add traefik 2 chart

### PR DESCRIPTION
**Pull request by joejulian**
Copy in the upstream chart: git@github.com:containous/traefik-helm-chart.git

The chart isn't served by them so this is for us to start with.

### CODE REVIEW

**Comment by GoelDeepak** [comment]
I personally have resistance to moving to Traefik 2.0 just yet. Maybe we end up with Traefik 2.0 but I want to run a comparison between Nginx, Traefik, and Envoy before making this move. The reason being ingress is an important component and any change would probably be a big undertaking both in terms of development and operation (upgrades and all). I also didn't like the direction that Traefik 2.0 took with custom CRDs which is a big departure from Kubernetes ingress APIs. I would like to know how Nginx and Envoy have taken

**Comment by joejulian** [comment]
This is just adding the chart for hosting. It's not modified from upstream in any way and I have no plans on moving forward with creating an addon at this time. I am going to do some testing of it on my home cluster though.

File: staging/traefik2/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Traefik
+
+[Traefik](https://traefik.io/) is a modern HTTP reverse proxy and load balancer made to deploy
+microservices with ease.
+
+**NOTE: THIS CHART IS IN INCUBATOR STATUS, DO NOT USE IN A PRODUCTION ENVIRONMENT**
```
Comment (samvantran) [inline staging/traefik2/README.md]: scary note

Comment (alejandroEsc) [reply staging/traefik2/README.md]: No addons will be created from this today. @joejulian wants to test this out. I am also using this chart as a test bed for a few things.

**Comment by jimmidyson** [comment]
Why can't we just use the chart from upstream repo if we're not changing it?

**Comment by joejulian** [comment]
It's not hosted anywhere and the way I wanted to work with it required it be hosted. If it's really a big deal, I can just host it myself.

**Pull request merged**